src/utils/Auth.js

```diff
@@ -0,0 +1,46 @@
+// const BASE_URL =
+// process.env.NODE_ENV === 'production'
+// ? 'https://api.arraktur-news-explorer.students.nomoredomainssbs.ru'
+// : 'http://localhost:3000';
+
+const BASE_URL =
+  'https://api.arraktur-news-explorer.students.nomoredomainssbs.ru';
+
+const handleResponse = (res) =>
+  res.ok ? res.json() : Promise.reject(`Error: ${res.status}`);
+
+const register = async (email, password, name) => {
+  const res = await fetch(`${BASE_URL}/signup`, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({ email, password, name }),
+  });
+  return handleResponse(res);
```

Comment:
**Might be better**

You can make a special method for fetching and checking responses **not to duplicate** it in every request:

```js

  function request(url, options) {
    return fetch(url, options).then(handleResponse)
  }

```

And now you can safely replace `fetch` with `request`. And you don’t need to change anything else.